ttnn/cpp/ttnn/operations/moreh/moreh_fold/device/fold_device_operation.cpp

```diff
@@ -0,0 +1,118 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "fold_device_operation.hpp"
+
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/tensor/types.hpp"
+
+namespace ttnn::operations::moreh::moreh_fold {
+void MorehFoldOperation::validate_inputs(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    auto& input = tensor_args.input;
+    auto input_shape = input.get_logical_shape().as_vector();
```

Comment:
Wouldn't it be okay to use shape instead of vector?

Reply:
resolved in latest commit.